lib/webauth.js

```diff
@@ -185,24 +185,38 @@ module.exports = function(authManager) {
         const authPluginSession = getAuthPluginSession(req, pluginID, {});
         req[`${UNP.APP_NAME}Data`].plugin.services = 
           authServiceHandleMaps[pluginID];
-        const wasAuthenticated = handler.getStatus(authPluginSession).authenticated;
-        let handlerResult;
         
-        //dont allow refresh if zlux cant find id or overall session expired
-        if (type === SESSION_ACTION_TYPE_REFRESH
+        const authHandlerStatus = handler.getStatus(authPluginSession);
+        const wasAuthenticated = authHandlerStatus.authenticated;
+        const hasGetCapabilities = !!((typeof handler.getCapabilities) === 'function');
+        let handlerResult;
+
+       if(!hasGetCapabilities){
+         authLogger.warn(`${pluginID}: getCapabilities() is not a function`);
+       }
+
+       if (type === SESSION_ACTION_TYPE_REFRESH
```

Comment:
This is technically correct, and also probably  this is an attempt to make a minimal change, but in order to make the intent clearer, I would make a bigger change:
1. Get rid of the `functionName` variable completely from this function: we only need `type` to track what's going on, no need to use two variables for that
2. rewrite it somehow like this: (NOT TESTED!!! review carefully)
```
       if (type === SESSION_ACTION_TYPE_REFRESH) {
         if (authManager.sessionTimeoutMs !== TIMEOUT_VALUE_NO_EXPIRE
            && (!timeout || timeout < Date.now())) {
           req.session.zlux = undefined;
           handlerResult = {success:false, reason:REASON_ZLUX_SESSION_EXPIRE};
         } else if (hasGetCapabilities && handler.getCapabilities().canRefresh) {
           handlerResult = yield handler.refreshStatus(req, authPluginSession);
         } else {
           handlerResult = { success: true };
         }        
       } else if (type === SESSION_ACTION_TYPE_AUTHENTICATE) {
         handlerResult = yield handler.authenticate(req, authPluginSession);
       } 
```
In my view, this is more straightforwardly presenting what we want to do here.

Reply:
Reviewed, implemented and tested.  Works perfectly @fkovinAtRocket